src/OpenApi/src/Services/Schemas/OpenApiSchemaService.cs

```diff
@@ -82,9 +82,9 @@ internal sealed class OpenApiSchemaService(
                 };
             }
             // STJ uses `true` in place of an empty object to represent a schema that matches
-            // anything. We override this default behavior here to match the style traditionally
-            // expected in OpenAPI documents.
-            if (type == typeof(object))
+            // anything (like the `object` type) or types with user-defined converters. We override
+            // this default behavior here to match the style traditionally expected in OpenAPI documents.
+            if (schema.GetValueKind() == JsonValueKind.True)
```

Comment:
Idle thought: what happens for `dynamic`?

Reply:
Good question! Looks like it gets handled similarily to `object foo`. I added test coverage for this in https://github.com/dotnet/aspnetcore/pull/57067/commits/fa39cfcfd1c32f410dc501a1e184f57539dcd144.